[Ee Zhi Jun Justin] iP

eezj35 opened the pull request:
# DukePro

> "Remember, I cannot" - someone who forgets things

Forget about forgetting things you need to do. DukePro helps you to manage your tasks!

It is:
* *text-based*
* *easy to use*
* ~~*FAST*~~ *SUPER FAST*

To use, just:
1. Type commands and tasks in the terminal
2. add your tasks.
3. let DukePro manage your tasks for you 😉

Best part is, **IT'S FREE**!

Features:
- [x] Define and add 3 types of tasks (events, todos, deadlines)
- [x] Mark tasks as done
- [x] See all your tasks
- [x] Delete tasks
- [x] Find tasks
- [ ] Reminders (coming soon)

Below is the `<main>` method, check out the full repo [here](https://github.com/eezj35/ip):
```java
public static void main(String[] args) {
        try {
            new Duke("data/duketest.txt").result();
        } catch (FileNotFoundException e) {
            ui.fileNotFoundMsg();
        }
    }
```

Commit:
Add Gradle support

eezj35 committed:
Level-1

eezj35 committed:
added StorageList.java
implemented Level-2

eezj35 committed:
created new Task class. Implemented level 3 - mark as done

eezj35 committed:
created ToDos.java, Events.java and Deadlines.java

eezj35 committed:
updated Task.java and its children classes

eezj35 committed:
implemented Level-4. updated children classes of Task.java and StorageList.java

eezj35 committed:
implemented TextUiTesting. edited Duke.java

eezj35 committed:
implemented level-5. added DukeException.java

eezj35 committed:
implemented level-6. added delete method in StorageList.java

eezj35 committed:
Classes with printing methods: add necessary indentations

eezj35 committed:
Add getter methods for additional variables

eezj35 committed:
Implement Level-7

Add readFile method to read from file and add tasks to list.
Add save method to save changes onto file after exiting.
Relevant Exceptions are also handled

eezj35 committed:
Implement Level-8

eezj35 committed:
Implement Level-7

Update save method to save changes while program still runs

eezj35 committed:
Edit all code to fit conventions

eezj35 committed:
Edit all code to fit conventions

eezj35 committed:
Implement Level-8

Take into account having to read different date and time inputs

eezj35 committed:
Implement Level-8

Clean up code for date and time parsing

eezj35 committed:
Merge branch 'branch-Level-7'

eezj35 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadlines.java
#	src/main/java/Duke.java
#	src/main/java/DukeException.java
#	src/main/java/Events.java
#	src/main/java/StorageList.java
#	src/main/java/Task.java
#	src/main/java/ToDos.java

eezj35 committed:
Implement  A-MoreOOP

Add Ui class to extract user interactions

eezj35 committed:
Implement  A-MoreOOP

Several related code can be extracted as classes.
StorageList class is equivalent to TaskList in question,
and is updated.

Let's,
* add Ui class to deal with user interactions
* add Storage class to deal with loading and saving
* add Parser to parse user input

eezj35 committed:
Implement  A-Packages

eezj35 committed:
Edit runtest.bat to fit code changes

EXPECTED.txt and input.txt also edited to fit new code

eezj35 committed:
Implement A-JUnit

eezj35 committed:
Update files after JUnit testing

eezj35 committed:
Implement A-JavaDoc

eezj35 committed:
Implement A-CodingStandard

eezj35 committed:
Implement Level-9

eezj35 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/StorageList.java
#	src/main/java/duke/Ui.java

eezj35 committed:
Merge remote-tracking branch 'origin/branch-Level-9'

eezj35 committed:
Implement A-CodingStandard

Remove unnecessary constructor from StorageList.java

eezj35 committed:
Merge branch 'branch-A-CodingStandard'

eezj35 committed:
Reset to branch-A-CodingStandard initial state

eezj35 committed:
Revert "Reset to branch-A-CodingStandard initial state"

This reverts commit 1113ea10f0df45b33e1c336d0eb8494811be3571.

eezj35 committed:
Merge branch 'branch-A-CodingStandard'

yucheng11122017 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,148 @@
+    /**
+     * Executes the program after reading input from file or user.
+     */
+    private void result() {
```
Maybe you could reduce the length of this function? 😄

zhuoyang125 replied:
Perhaps this method name could be changed to a verb to make it more descriptive?

eezj35 replied:
Ok i will go try and shorten the length of the function and change the name of the method!

yucheng11122017 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,148 @@
+    private static final int TASK_TODO = 1;
+    private static final int TASK_DEADLINE = 2;
+    private static final int TASK_EVENT = 3;
+    private static final int TASK_OUTOFBOUNDS = 4;
```
Maybe it would be better to not name the two invalid conditions (out of bounds and unknown) with task as a prefix as well? Because I initially taught it was a type of task 😢

eezj35 replied:
Ah true can be quite confusing! i'll change this

yucheng11122017 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,13 @@
+package duke;
+
+public class DukeException extends Exception{
```
Don't forget the white space before the curly brackets 😸

eezj35 replied:
Thanks for spotting that!

yucheng11122017 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+
+public class Parser {
+
+    public Parser() {
```
Could it be better to delete this function? 🤔

eezj35 replied:
hahah it is quite useless, thanks for pointing it out!

yucheng11122017 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+     * @param input Done command.
+     * @return Whether index is valid integer.
+     */
+    public boolean isInteger(String input) {
```
Perhaps you could make the name a bit more specific as to what this is for. 😸

eezj35 replied:
Mm true, it can be quite confusing. Thanks!

yucheng11122017 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,118 @@
+
+public class Ui {
+
+    public Ui() {}
```
Maybe you could consider deleting this. 😸

yucheng11122017 commented on src/main/java/duke/task/Deadlines.java:
```diff
@@ -0,0 +1,47 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task {
```
Shouldn't this class be in singular form since it is only referring to one deadline? 😄

zhuoyang125 replied:
Yep I agree. The class may be misunderstood as a collection of Task objects, so it might be better to use the singular form.

eezj35 replied:
oh yea that is true, thanks for pointing this out!

yucheng11122017 commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,46 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Events extends Task {
```
Shouldn't this be in singular form?

yucheng11122017 commented on src/main/java/duke/task/ToDos.java:
```diff
@@ -0,0 +1,26 @@
+package duke.task;
+
+public class ToDos extends Task {
```
Once again, maybe this should be singular form? 😄

yucheng11122017 reviewed (commented):
Hi Justin! Good job on this 😄 
Hope you find my comments useful!

zhuoyang125 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+    public LocalDateTime parseLocalDateTime(String time) {
+        DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+        if (time.substring(0, 1).matches("[0-9]+")) {
+            DateTimeFormatter anotherDTF = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Should the variable 'anotherDTF' be named as 'anotherDtf' instead?

eezj35 replied:
ah thanks for the spot!

zhuoyang125 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+     * @return Event description.
+     */
+    public String getEventDescription(String input) {
+        return input.substring(6,  input.indexOf("/") - 1);
```
A really minor issue but there seems to be an additional white space after the comma

eezj35 replied:
thanks for the spot!

zhuoyang125 commented on src/main/java/duke/StorageList.java:
```diff
@@ -0,0 +1,109 @@
+                    storageList.add(dl);
+                } else if (type == 'E') {
+                    Events event = new Events(taskDesc, taskTime);
+                    if(doneState == 1){
```
Perhaps white spaces could be added here to improve readability

zhuoyang125 commented on src/main/java/duke/StorageList.java:
```diff
@@ -0,0 +1,109 @@
+     *
+     * @param task
+     */
+    public void add(Task task){
```
Maybe consider making the method names a bit more descriptive? (e.g. 'addTask' instead of 'add')

zhuoyang125 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,118 @@
+     * @param i The case number.
+     * @return The error message relevant to the problem.
+     */
+
```
Perhaps this blank line could be removed?

zhuoyang125 reviewed (commented):
The code was easy to follow, and I like that the header comments were clear in describing the methods. I have some small suggestions to provide but overall, good job in following the coding standard!

eezj35 committed:
Correct coding standards, based on peer reviews

eezj35 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

eezj35 committed:
Edit gradle mainClassName

eezj35 committed:
Merge branch 'branch-A-Gradle'

eezj35 committed:
Implement Level-10

Incomplete version of GUI

eezj35 committed:
Update Level-10 implementation

eezj35 committed:
Shorten image URL to tinyurl

eezj35 committed:
Implement Level-10

eezj35 committed:
Make window and image size bigger

eezj35 committed:
Implement Level-10

eezj35 committed:
Merge branch 'branch-Level-10'

eezj35 committed:
Merge branch 'master' of https://github.com/eezj35/ip

* 'master' of https://github.com/eezj35/ip:
  Shorten image URL to tinyurl

# Conflicts:
#	src/main/java/duke/Duke.java

eezj35 committed:
Implement A-Assertions

This will help us to test the correctness of certain
assumptions made in the code, and thus help us to spot bugs.

Let's insert relevant assert statements in the places shown
in the commit, which have assumptions to be tested.

eezj35 committed:
Implement A-CodeQuality

eezj35 committed:
Merge pull request #1 from eezj35/branch-A-Assertions

Implement A-Assertions

eezj35 committed:
Merge branch 'master' of https://github.com/eezj35/ip

eezj35 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/DialogBox.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/StorageList.java
#	src/main/java/duke/task/Event.java

eezj35 committed:
Merge pull request #2 from eezj35/branch-A-CodeQuality

Implement A-CodeQuality

eezj35 committed:
Merge branch 'master' of https://github.com/eezj35/ip

eezj35 committed:
Delete unnecessary line from Duke.java

eezj35 committed:
Implement B-ViewSchedules

eezj35 committed:
Remove unnecessary assert statement

eezj35 committed:
Remove unnecessary throw declaration in viewSchedule method

eezj35 committed:
Added infinite minHeight property to Label

eezj35 committed:
Merge branch 'branch-BCD-Extension'

eezj35 committed:
Corrected assert statement

eezj35 committed:
Implement  A-BetterGui

eezj35 committed:
Finalize features and tidy up code

eezj35 committed:
Finalize more features

eezj35 committed:
Finalize features and edit code style

eezj35 committed:
Add relevant files needed for proper functionality

eezj35 committed:
Add files via upload

Screenshot of product

eezj35 committed:
Add product title

eezj35 committed:
Delete Ui.png

eezj35 committed:
Merge branch 'master' of https://github.com/eezj35/ip

eezj35 committed:
Add product screenshot

eezj35 committed:
Refine UI for marking as done

eezj35 committed:
Refine UI for marking task as done

eezj35 committed:
Update README.md

eezj35 committed:
Change mainClassName

eezj35 committed:
Update README.md

eezj35 committed:
Merge branch 'master' of https://github.com/eezj35/ip

eezj35 committed:
Update README.md

eezj35 committed:
Update README.md

eezj35 committed:
Update README.md

eezj35 committed:
Set theme jekyll-theme-cayman

eezj35 committed:
Set theme jekyll-theme-cayman

eezj35 committed:
Update README.md

eezj35 committed:
Implement auto creation of duketest text file
and data folder upon first time start up

eezj35 committed:
Merge branch 'master' of https://github.com/eezj35/ip
